src/Symfony/Component/Security/Core/Authorization/ExpressionLanguage.php

```diff
@@ -29,8 +29,10 @@ class_exists(ExpressionLanguageProvider::class);
      */
     class ExpressionLanguage extends BaseExpressionLanguage
     {
-        public function __construct(?CacheItemPoolInterface $cache = null, array $providers = [])
+        public function __construct(?CacheItemPoolInterface $cache = null, iterable $providers = [])
         {
+            $providers = \iterator_to_array($providers);
```

Comment:
Can you explain why you think we need to change the argument type if we then need to transform it to an array anyway?

Reply:
tldr:  for convenience

when you need to override the `Security\ExpressionLanguage` service and inject it with `tagged_iterator`, you have to introduce a child class _just_ to override the constructor, and cannot use the class itself.

same was true for the base `Component\ExpressionLanguage` until recently, when it got the constructor changed.

it has nothing to do with inner handling of the argument, the purpose is to eliminate the need for subclassing just to be able to use iterator in DI.